bot/exts/moderation/infraction/_scheduler.py

```diff
@@ -155,6 +170,8 @@ async def apply_infraction(
                 if expiry:
                     # Schedule the expiration of the infraction.
                     self.schedule_expiration(infraction)
+            except discord.NotFound:
+                log.info(f"Can't apply {infraction['type']} to user {infraction['user']} because user left from guild.")
```

Comment:
This can be merged with the 10007 check by checking the status as mentioned in another comment.